knowledge_repo/app/index.py

```diff
@@ -124,7 +124,34 @@ def index_up_to_date():
             return False
     return True
 
+@ErrorLog.logged
+def update_index_for_post(kp,path):
+    import time
+    is_index_master = acquire_index_lock()
+    
+    try:
+        IndexMetadata.set('lock', 'index', LOCKED)
+        db_session.commit()
```

Comment:
@abhirathb Is this required at this moment?

Reply:
Yep. This has been picked from the update_index code that they have written otherwise. I'v just removed the steps that filter out which posts to update, thus saving time